plugin/pkg/global/resourcereferencemanager/admission_test.go

```diff
@@ -529,6 +571,36 @@ var _ = Describe("resourcereferencemanager", func() {
 
 				Expect(err).To(HaveOccurred())
 			})
+
+			It("should reject because the user is not allowed to read the referenced resource", func() {
+				Expect(gardenCoreInformerFactory.Core().InternalVersion().CloudProfiles().Informer().GetStore().Add(&cloudProfile)).To(Succeed())
+				Expect(gardenCoreInformerFactory.Core().InternalVersion().Seeds().Informer().GetStore().Add(&seed)).To(Succeed())
+				Expect(gardenCoreInformerFactory.Core().InternalVersion().SecretBindings().Informer().GetStore().Add(&secretBinding)).To(Succeed())
+				Expect(kubeInformerFactory.Core().V1().ConfigMaps().Informer().GetStore().Add(&configMap)).To(Succeed())
+
+				user := &user.DefaultInfo{Name: "disallowed-user"}
+				attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, user)
+
+				err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+				Expect(err).To(HaveOccurred())
```

Comment:
```suggestion
				Expect(err).To(MatchError("shoot cannot reference a resource you are not allowed to read"))
```

Reply:
`Expect(err).To(HaveOccurred())` is the way errors are checked in this test (and many similar tests). I think checking specific error messages makes tests more brittle, as such messages may change. I would also keep error checking consistent throughout the file.